packages/syft/src/syft/service/attestation/utils.py

```diff
@@ -0,0 +1,82 @@
+# stdlib
+import base64
+
+# third party
+from cryptography.x509 import load_der_x509_certificate
+import jwt
+from jwt.algorithms import RSAAlgorithm
+import requests
+from result import Err
+from result import Ok
+from result import Result
+
+
+def verify_attestation_report(token: str) -> Result[Ok[dict], Err[str]]:
+    """
+    Verifies a JSON Web Token (JWT) using a public key obtained from a JWKS (JSON Web Key Set) endpoint.
+    The function handles two distinct processes for token verification depending on the token type:
+
+    - 'cpu': Fetches the JWKS from the 'jku' URL specified in the JWT's unverified header,
+             finds the key by 'kid', and converts the JWK to a PEM format public key for verification.
+
+    - 'gpu': Directly uses a fixed JWKS URL to retrieve the keys, finds the key by 'kid', and uses the
+             'x5c' field to extract a certificate which is then used to verify the token.
+
+    Parameters:
+        token (str): The JWT that needs to be verified.
+
+    Returns:
+        Result[Ok[dict], Err[str]]: A Result object containing the payload of the verified token if successful,
+                                    or an Err object with an error message if the verification fails.
+    """
+    try:
+        # Attempt to retrieve 'jku' from the unverified header to determine the JWKS URL
+        unverified_header = jwt.get_unverified_header(token)
+        jwks_url = unverified_header.get("jku", "")
+        token_type = "cpu" if jwks_url else "gpu"
```

Comment:
Here probably , we would need a reference, to Azure Attestation Key server, like we do nvidia, to verify if it was actually give by an Azure attestation Service.

But we could do that, when we are about to merge the PR, but this is good in the short term.

Reply:
Added a TODO for this

Reply:
Resolved the TODO in this PR itself